tests/test_apiclient.py

```diff
@@ -504,6 +504,20 @@ def test_find_users_by_supplier_id(self, data_client, rmock):
 
         assert user == self.user()
 
+    def test_find_users_not_possible_with_supplier_id_and_role(self, data_client, rmock):
+        with pytest.raises(ValueError):
+            data_client.find_users(supplier_id=123, role='buyer')
+
+    def test_find_users_by_role(self, data_client, rmock):
+        rmock.get(
+            "http://baseurl/users?role=buyer",
+            json=self.user(),
+            status_code=200)
+
+        user = data_client.find_users(role='buyer')
+
```

Comment:
Seems like all our other tests are checking that the mock has been called:

`assert rmock.called`